mne/source_tfr.py

```diff
@@ -0,0 +1,305 @@
+# -*- coding: utf-8 -*-
+#
+# Authors: Dirk Gütlin <dirk.guetlin@stud.sbg.ac.at>
+# TODO: This file uses a lot of stuff from io/base.py and source_estimate.py
+# TODO: Maybe name these persons as authors too?
+#
+# License: BSD (3-clause)
+
+import copy
+import numpy as np
+
+from .filter import resample
+from .utils import (_check_subject, verbose, _time_mask)
+from .io.base import ToDataFrameMixin, TimeMixin
+from .externals.h5io import write_hdf5
+
+
+class SourceTFR(ToDataFrameMixin, TimeMixin):
+    """Class for time-frequency transformed source level data.
+
+    Parameters
+    ----------
+    data : array, shape (n_dipoles, n_freqs, n_times) | tuple, shape (2,)
+        Time-frequency transformed data in source space. The data can either
+        be a single array or a tuple with two arrays: "kernel" shape
+        (n_vertices, n_sensors) and "sens_data" shape (n_sensors, n_freqs,
+        n_times). In this case, the source space data corresponds to
+        "numpy.dot(kernel, sens_data)".
+    vertices : array | list of array
+        Vertex numbers corresponding to the data.
+    tmin : float
+        Time point of the first sample in data.
+    tstep : float
+        Time step between successive samples in data.
+    subject : str | None
+        The subject name. While not necessary, it is safer to set the
+        subject parameter to avoid analysis errors.
+    %(verbose)s
+
+    Attributes
+    ----------
+    subject : str | None
+        The subject name.
+    times : array, shape (n_times,)
+        The time vector.
+    vertices : array | list of array of shape (n_dipoles,)
+        The indices of the dipoles in the different source spaces. Can
+        be an array if there is only one source space (e.g., for volumes).
+    data : array of shape (n_dipoles, n_times)
+        The data in source space.
+    shape : tuple
+        The shape of the data. A tuple of int (n_dipoles, n_times).
+    """
+
+    @verbose
+    def __init__(self, data, vertices=None, tmin=None, tstep=None,
```

Comment:
break the `__init__` with functions. I know that some of our inits are huge, but they should be short easy to read in a glance.

```py
def __init__(foo, bar, data, vertices,  bla):
  _check_param(foo, ...)
  self._data = _prepare_data(data)
  self._vercies = _foo(vertices)
  ...
```

to me init should be just assigning stuff to the attributes. If `_prepare_data` has a signature like `_prepare_data(data=data, n_col=vertices.shape[0], moon_color=foo)` it tels me at a glance how data is related to the other things. Sure I would not know the details but I least I would know the possible relations. If I have to read the code and strip all the value errors out on my mind the task of reading the code is much harder.

keep in mind: **code is written once but read multiple times** and when you read code you want to skim it. Not digest it.